src/app/qgsdecorationitem.h

```diff
@@ -37,6 +37,14 @@ class APP_EXPORT QgsDecorationItem: public QObject
       BottomRight,
     };
 
+    //! Placement units
+    enum MarginUnits
```

Comment:
Instead of creating a new enum, it would be better to reuse QgsSymbolV2::OutputUnit and add the "Percentage" mode to that enum. In any case Percentage will eventually need to be added to that enum for both labeling and layer effects, so may as well do that now and avoid the duplicate code.